store/retry.go

```diff
@@ -70,6 +71,9 @@ func shouldRetryError(attempt *retry.Attempt, err error) bool {
 	if !attempt.More() {
 		return false
 	}
+	if urlErr, ok := err.(*url.Error); ok {
+		return urlErr.Err == io.ErrUnexpectedEOF || urlErr.Err == io.EOF
```

Comment:
Good idea, done, althouth it needs to be urlErr.Err, and also the follow up condition on netErr needs "else if", otherwise the re-assigned err falls into it and into Timeout() check.